sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/TopicSubscriptionWithRuleOperationsSample.java

```diff
@@ -0,0 +1,210 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Sample demonstrates how to receive {@link ServiceBusReceivedMessage messages} by topic subscriptions from
+ * an Azure Service Bus Topic.
+ */
+public class TopicSubscriptionWithRuleOperationsSample {
+
+    static final String SERVICE_BUS_CONNECTION_STRING = System.getenv("AZURE_SERVICEBUS_NAMESPACE_CONNECTION_STRING");
+    static final String SERVICE_BUS_TOPIC_NAME = System.getenv("AZURE_SERVICEBUS_SAMPLE_TOPIC_NAME");
+    static final String ALL_MESSAGES_SUBSCRIPTION_NAME = "{Subscription 1 Name}";
+    static final String SQL_FILTER_ONLY_SUBSCRIPTION_NAME = "{Subscription 2 Name}";
+    static final String SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME = "{Subscription 3 Name}";
+    static final String CORRELATION_FILTER_SUBSCRIPTION_NAME = "{Subscription 4 Name}";
+    static final String DEFAULT_SUBSCRIPTION_RULE_NAME = "$Default";
+    static final String SQL_FILTER_ONLY_SUBSCRIPTION_RULE_NAME = "RedSqlRule";
+    static final String SQL_FILTER_WITH_ACTION_SUBSCRIPTION_RULE_NAME = "BlueSqlRule";
+    static final String CORRELATION_FILTER_SUBSCRIPTION_RULE_NAME = "ImportantCorrelationRule";
+
+    /**
+     * Main method to invoke this demo on how to receive {@link ServiceBusReceivedMessage messages}
+     * by topic subscriptions from an Azure Service Bus Topic.
+     *
+     * @param args Unused arguments to the program.
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    public static void main(String[] args) throws InterruptedException {
+        TopicSubscriptionWithRuleOperationsSample app = new TopicSubscriptionWithRuleOperationsSample();
+        app.run();
+    }
+
+    /**
+     * This method to invoke this demo on how to receive {@link ServiceBusReceivedMessage messages}
+     * by topic subscriptions from an Azure Service Bus Topic.
+     *
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    void run() throws InterruptedException {
+
+        ServiceBusAdministrationClient administrationClient =
+            new ServiceBusAdministrationClientBuilder()
+                .connectionString(SERVICE_BUS_CONNECTION_STRING)
+                .buildClient();
+
+        // First Subscription is already created with default rule. Leave as is.
+        System.out.println(String.format("SubscriptionName: %s, Removing and re-adding Default Rule", ALL_MESSAGES_SUBSCRIPTION_NAME));
+        administrationClient.deleteRule(SERVICE_BUS_TOPIC_NAME, ALL_MESSAGES_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME);
+        administrationClient.createRule(SERVICE_BUS_TOPIC_NAME, ALL_MESSAGES_SUBSCRIPTION_NAME, DEFAULT_SUBSCRIPTION_RULE_NAME);
```

Comment:
Instead of creating the rule and then updating it, we should use the overload that takes CreateRuleOptions.
https://azuresdkdocs.blob.core.windows.net/$web/java/azure-messaging-servicebus/7.2.3/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClient.html#createRule-java.lang.String-java.lang.String-java.lang.String-com.azure.messaging.servicebus.administration.models.CreateRuleOptions-

Reply:
This applies to all other instances where you do the same pattern.

Reply:
According to your comment, fixed within new version